packages/shared/lib/sdk/sync.ts

```diff
@@ -374,6 +385,19 @@ export class NangoAction {
         return this.nango.getMetadata(this.providerConfigKey as string, this.connectionId as string);
     }
 
+    public async getWebhookURL(): Promise<string> {
+        const webhookBaseUrl = await getGlobalWebhookReceiveUrl();
+        const providerConfigKey: string = this.providerConfigKey as string;
+        const response = await this.nango.getIntegration(providerConfigKey);
+        if (!response || !response.config || !response.config.provider) {
+            throw Error(`There was no provider found for the provider config key: ${providerConfigKey}`);
+        }
+        const environmentUuid = await environmentService.getAccountUUIDFromEnvironment(this.environmentId as number);
+        const webhookURL = `${webhookBaseUrl}/${environmentUuid}/${response.config.provider}`;
+
+        return webhookURL;
```

Comment:
this is a similar logic as the one in InternalNango. Anyway we could consolidate them?

Reply:
Yeah ideally the in internalNango can go away actually. I’ll remove it from there